Tasks/Gradle/CodeAnalysis/pmd.gradle

```diff
@@ -0,0 +1,31 @@
+rootProject {
+    afterEvaluate {project ->
+        def hasPmd = false;
+
+        allprojects.forEach{ p ->
+
+            if (p.plugins.hasPlugin('pmd')) {
+                hasPmd = true;
+                println('$p is already configured to run PMD. Skipping auto configuration ');
+            }
+        }
+
+        if (!hasPmd) {
+            println('Applying the PMD plugin to all projects');
+            allprojects.forEach{p->
+                p.apply plugin: 'pmd'
+            }
+        }
+
+        allprojects { p->
+            if (!hasPmd) {
```

Comment:
Does the _!hasPmd_ variable test have to be done for each project, or could it done once outside the loop?
Also, could the configuration step be combined with the code above in lines 13-18?